File: pset6/credit/credit.py
from cs50 import get_int

def main():

    while True:
        num = get_int("Credit Card Number: ")
        if num > 0:
            break

    digits = 0
    checksum = 0

    for digits in range(num, 0, 1):
        if (digits % 2) == 0:
            checksum += (num % 10)
        else:
            odd = ((num % 10) * 2)
            if odd > 9:
                checksum += ((odd / 10) + (odd % 10))
            else:
                checksum += odd
        num /= 10

    checksum %= 10

    if checksum == 0 and ((num >= 4e12 and num < 5e12) or (num >= 4e15 and num < 5e15)):
        print("VISA")
    elif checksum == 0 and (num >= 51e14 and num < 56e14):
        print("MASTERCARD")
    elif (checksum == 0) and ((num >= 34e13 and num < 35e13) or (num >= 37e13 and num < 38e13)):
        print("AMEX")
    else:
        print("INVALID")

main()

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //Prompt the user for credit card number
    const long num;
    do
    {
        num = get_long("Credit Card Number: ");
    }
    while (num < 0);

    int digits, checksum = 0;
    
    //Calculate the number of digits in the number
    for (digits = 0; num > 0; digits++, num /= 10)
    {
        //While calculating, determine the digits in the
        //Even positions and add them to the checksum
        if (digits % 2 == 0)
        {
            checksum += num % 10;
        }
        //Do the same for the odd digits multiplying them by 2
        else
        {
            int odd = (num % 10) * 2;
            //Separate the digits in the number and add them separately
            if (odd > 9)
            {
                checksum += ((odd / 10) + (odd % 10));
            }
            else
            {
                checksum += odd;
            }
        }

    }
    checksum %= 10;
 
    //Conditions for each card
    if ((checksum == 0) && ((number >= 4e12 && number < 5e12) || (number >= 4e15 && number < 5e15)))
    {
        printf("VISA\n");
    }
    else if ((checksum == 0) && (number >= 51e14 && number < 56e14))
    {
        printf("MASTERCARD\n");
    }
    else if ((checksum == 0) && ((number >= 34e13 && number < 35e13) || (number >= 37e13 && number < 38e13)))
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
}